File: 2.py
def is_palindrome(x: int) -> bool:
    if x < 0:
        return False
    if x != 0 and x % 10 == 0:
        return False
    y = x
    m = 0
    while x > 0:
        n = x % 10
        m = m * 10 + n
        x //= 10
    return m == y

File: 2_Palindrome/2.js
var isPalindrome = function(x) {
    if (x < 0) return false;  // negatives aren't palindromes
    if (x % 10 === 0 && x !== 0) return false;  // trailing zero (except 0 itself)
    
    let str_x = x.toString();
    let str_x_reversed = str_x.split('').reverse().join('');
    return str_x === str_x_reversed;
};

console.log(isPalindrome(121));    // true
console.log(isPalindrome(-121));   // false
console.log(isPalindrome(10));     // false